src/DataGrid.tsx

```diff
@@ -940,32 +950,40 @@ function DataGrid<R, SR, K extends Key>(
     return isDraggedOver ? selectedPosition.idx : undefined;
   }
 
-  function getSelectedCellProps(rowIdx: number): SelectedCellProps | EditCellProps<R> | undefined {
-    if (selectedPosition.rowIdx !== rowIdx) return;
+  function getSelectedCellProps(rowIdx: number): SelectedCellProps<R, SR> | undefined {
```

Comment:
Could destructure `selectedPosition` so we don't repeat property access on `idx` and `rowIdx`